### PR TITLE
CMake+vcpkg GitHub workflows

### PR DESCRIPTION
**Pull request by theblackunknown**

### CODE REVIEW

**Comment by JonathanSalwan** [comment]
Ping me when it's ok for a review (it's currently as draft MR).

**Comment by theblackunknown** [comment]
> Ping me when it's ok for a review (it's currently as draft MR).

I made good progress but I am still struggling at the install or test steps if I remember correctly. 
Things are slowing down on my side as this is the holiday season :)

**Comment by JonathanSalwan** [comment]
Do not worry, no pressure, life is to short too spending it on vcpkg so enjoy your holidays :D.

**Comment by theblackunknown** [comment]
Hey @JonathanSalwan I am mostly done on this PR I think.
I got a CMake+vcpkg Github workflow set up and running fine for : build, installation and artifact upload.

I am experimenting on whether I can also add tests using `ctest` but not sure if I will end up successful on all platform.
I see that tests in `src/testers` requires dependencies not easily prepared for Windows/macOS platforms.

**Comment by theblackunknown** [comment]
I have rebased and split the work in small and meaningful commits, feel free to review them one by one if that makes the review process easier.

**Comment by JonathanSalwan** [comment]
> I got a CMake+vcpkg Github workflow set up and running fine for : build, installation and artifact upload.

Awesome!

> I am experimenting on whether I can also add tests using ctest but not sure if I will end up successful on all platform.
> I see that tests in src/testers requires dependencies not easily prepared for Windows/macOS platforms.

Do not worry, Github workflows are like tests.

**Comment by JonathanSalwan** [comment]
Salut,

Je viens de tester et il semblerait que maintenant avec un `make check` on process 2 fois les unittest python:

```
$ make -C build check
make: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[1]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[2]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 87%] Built target triton
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 89%] Built target ctest_api
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 89%] Built target python-triton
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 528 tests in 348.941s

OK
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 89%] Built target test-python
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 91%] Built target taint_reg
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 93%] Built target info_reg
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 95%] Built target ir
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[ 97%] Built target simplification
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
[100%] Built target constraint
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
Test project /home/jonathan/Works/Tools/Triton/build
      Start  1: test-python
      [...]
```

**Comment by theblackunknown** [comment]
> Salut,
> 
> Je viens de tester et il semblerait que maintenant avec un `make check` on process 2 fois les unittest python:
> 
> ```
> $ make -C build check
> make: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[1]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[2]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 87%] Built target triton
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 89%] Built target ctest_api
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 89%] Built target python-triton
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> ................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
> ----------------------------------------------------------------------
> Ran 528 tests in 348.941s
> 
> OK
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 89%] Built target test-python
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 91%] Built target taint_reg
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 93%] Built target info_reg
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 95%] Built target ir
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [ 97%] Built target simplification
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> [100%] Built target constraint
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Leaving directory '/home/jonathan/Works/Tools/Triton/build'
> make[3]: Entering directory '/home/jonathan/Works/Tools/Triton/build'
> Test project /home/jonathan/Works/Tools/Triton/build
>       Start  1: test-python
>       [...]
> ```

Yo !

En effet, j'ai ajouté certains tests via `add_test` car j'ai migré l'exécution des tests avec `ctest` directement. 
Sauf que je n'ai pas retiré les `add_custom_target` existantes, qui duplique certains tests. 
Par exemple il existe un `add_custom_target(test-python ...)` et `add_test(test-python ...)` qui font la même chose. 
Or vu que `check` dépends de ces custom target et lance ctest, ces tests sont exécutés deux fois. 

tldr: on devrait pouvoir simplement retirer les `add_custom_target` redondantes. Veux-tu que je fasse une PR pour ça ?

**Comment by JonathanSalwan** [comment]
> tldr: on devrait pouvoir simplement retirer les add_custom_target redondantes. Veux-tu que je fasse une PR pour ça ?

Yes je veux bien :)

**Comment by JonathanSalwan** [comment]
Au fait, est-ce que tu peux vérifier que la section [Install Windows](https://github.com/JonathanSalwan/Triton/tree/dev-v1.0#windows) dans le readme est toujours d'actualité avec ta contrib ?
